service/client_state.hh

```diff
@@ -136,6 +144,10 @@ private:
     // Only populated for external client state.
     auth::service* _auth_service{nullptr};
 
+    // For restoring default values in the timeout config
```

Comment:
I don't understand this comment. Why do we need both these timeout configs, and why is the "default" per connection as well?

Reply:
Let's assume for a while that we already have a per-service-level timeout mechanism from #7913. Now, a user can do:
```cql
ALTER SERVICE LEVEL sl SET read_timeout = 5s;
```
and later:
```cql
ALTER SERVICE LEVEL sl SET read_timeout = null;
```
What the second statement does is removing the custom timeout value, and in this case the session is expected to fall back to its default timeout configuration, which is the one defined in `scylla.yaml`. In order to avoid having to consult database config, a local copy of default values is kept.

Reply:
I think this change is not strictly needed for this pull request though, so I can remove it for now. It's only needed in the broader context of #7913.

Reply:
Could we not merge _default_timeout_config into _timeout_config, so we only need one? So the lookup is fast.

Reply:
Which lookup? We only use `_timeout_config` for getting proper timeout values. `_default_timeout_config` is here in order to know what to put in `_timeout_config` in case somebody removes a custom timeout value from their config (e.g. by updating their service level parameters). When they do that, this value is set to its corresponding value from `_default_timeout_config`.